api/courses.js

```diff
@@ -321,4 +317,66 @@ router.get("/:id/assignment", async (req, res) => {
   }
 });
 
+router.get(
+  "/:id/roster",
+  auth.authenticate,
+  auth.authorize([ROLES.admin, ROLES.instructor]),
+  async (req, res) => {
+    try {
+      const course = await coursesModel.getCourseById(req.params.id);
+      if (course) {
+        if (
+          (req.role == "instructor" && course.instructorId == req.userId) ||
```

Comment:
I do have a function for checking if a instructor is authroized to access a course - and here you could use === ROLES.instructor if we wanted

Reply:
async function authorizeCourseInstructor(instructorId, courseId) {
  const course = await getCourse(courseId);
  if (!course) throw new Error("Request body is not a valid assignment object");
  return course.instructorId.toString() === instructorId;
}

Reply:
so maybe if (req.role === ROLES.instructor) {
authorized? = authroizeCoursInstructor .....
...
}
if (authorized) ---- 

Dunno if that's any cleaner or anything lol but idk hah

Reply:
Thanks, I updated the code to use this!